core-java-modules/core-java-collections-set-2/src/test/java/com/baeldung/cartesianproduct/CartesianProductUnitTest.java

```diff
@@ -0,0 +1,93 @@
+package com.baeldung.cartesianproduct;
+
+import static org.testng.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.testng.annotations.Test;
+
+public class CartesianProductUnitTest {
+    private CartesianProduct cp = new CartesianProduct();
+    List<List<Object>> sets = Arrays.asList(
+        Arrays.asList(10, 20),
+        Arrays.asList("John", "Jack"),
+        Arrays.asList('I', 'J')
+    );
+
+    @Test
+    public void shouldCalculateCartesianProductUsingStreams() {
```

Comment:
Please, follow the guidelines for [writing tests](https://docs.google.com/document/d/1iIeSPuYFG-jEtB8UbcxQnn2AI3OR2LWlnisNo2NUCL8/edit) for test method naming and content

Reply:
and apply for all test methods

Reply:
Somehow I missed this information, thank you for pointing it out. It's fixed now.